connector/mqtt/vernemq/README.md

```diff
@@ -0,0 +1,301 @@
+
+![image](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/dojot/dojot/epic-100kMqttDevices/connector/mqtt/vernemq/docs/plant_uml/mqtt/diag_mqtt)
+
+Fig. 1 - VerneMQ with Dojot (Whereas V2K, K2V, VerneMQ and Client  already have certificates, can occur later communications between PKI and these services. This is better explained later.)
+
+## V2K-bridge
+
+The V2K-bridge service is the core between the communication of VerneMQ broker and the Kafka broker.
+
+For more see [here](./v2k-bridge)
+
+## K2V-bridge
+
+The k2v-bridge service is who consume messages from kafka, extract its attrs value, build a topic with the kafka key where the message where produced, then publish the attrs payload to verneMQ.
+
+For more see [here](./k2v-bridge)
+
+## MQTT Security: VerneMQ + TLS + PKI (EJBCA) + ACL
+
+We use **TLS** mutual authentication to provide a secure communication between devices and broker ([VerneMQ](https://github.com/vernemq/vernemq)) through **MQTT**.  Transport Layer Security (**TLS**) is a cryptographic protocol designed to provide communications security over a computer network.
+
+In **TLS** each device and broker should be provisioned with a private key and corresponding public certificate sign from **CA** (certificate authority) and a root certificate (public certificate of **CA**), the **CA** is include in **PKI** (public key infrastructure). A **PKI**  ([EJBCA](./ejbca)) is a system for the creation, storage, and distribution of digital certificates.
+
+An **ACL** (access-control list) based authorization is provided to manage permissions, so a device can only publish and subscribe to its own topic, the topics are: 
+
+- For publish: ***tenant***_:_***device_id***_/attrs_
+- For subscription: ***tenant***_:_***device_id***_/config_
+
+Where ***tenant*** is the information context separation of dojot and ***device_id*** is a unique identification for the device.  
+The junction (***tenant:device_id***) of *tenant* and  *device_id* must be unique.
+
+See more about [ACL Plugin](vernemq/plugins/dojot_acl_plugin) for VerneMQ.
+
+Also a PKI includes the certificate revocation list (**CRL**), which is a list of certificates that have been revoked before reaching the expiration date of the certificate.
+
+The process of obtaining certificates for client (Fig. 2):
+
+- Creation of entity at EJBCA, usually this unique entity consists of ***tenant***_:_***device_id***.
+
+- Obtaining a public certificate for the client, which happens with the creation of the CSR (Certificate Signing Request, it usually contains the public key, identifying information such as a hostname and unique identification) and its submission to obtain a public certificate.
+
+- Obtaining the root certificate (public certificate of **CA**).
+
+![image](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/dojot/dojot/epic-100kMqttDevices/connector/mqtt/vernemq/docs/plant_uml/mqtt/seq_sec_client)
```

Comment:
You can see the image before merge here
http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/mprevide/dojot/readme_mqtt/connector/mqtt/vernemq/docs/plant_uml/mqtt/seq_sec_client